exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpHttpSpanExporter.java

```diff
@@ -0,0 +1,211 @@
+
+    CompletableResultCode result = new CompletableResultCode();
+
+    client
+        .newCall(requestBuilder.build())
+        .enqueue(
+            new Callback() {
+              @Override
+              public void onFailure(@NotNull Call call, @NotNull IOException e) {
+                spansExportedFailure.add(spans.size());
+                result.fail();
+                logger.log(
+                    Level.SEVERE,
+                    "Failed to export spans. The request could not be executed. Full error message: "
+                        + e.getMessage());
+              }
+
+              @Override
+              public void onResponse(@NotNull Call call, @NotNull Response response) {
+                if (response.isSuccessful()) {
+                  spansExportedSuccess.add(spans.size());
+                  result.succeed();
+                  return;
+                }
+
+                spansExportedFailure.add(spans.size());
+                result.fail();
+                int code = response.code();
+
+                Status status;
+                try {
+                  status = requestResponseHandler.extractErrorStatus(response.body());
+                } catch (IOException e) {
+                  status =
+                      Status.newBuilder()
+                          .setMessage(
+                              "Unable to extract error message from request: " + e.getMessage())
+                          .build();
+                }
+
+                logger.log(
+                    Level.SEVERE,
+                    "Failed to export spans. Server responded with code "
+                        + code
+                        + ". Error message: "
+                        + status.getMessage());
+              }
+            });
+
+    return result;
+  }
+
+  /**
+   * The OTLP exporter does not batch spans, so this method will immediately return with success.
+   *
+   * @return always Success
+   */
+  @Override
+  public CompletableResultCode flush() {
+    return CompletableResultCode.ofSuccess();
+  }
+
+  /**
+   * Returns a new builder instance for this exporter.
+   *
+   * @return a new builder instance for this exporter.
+   */
+  public static OtlpHttpSpanExporterBuilder builder() {
+    return new OtlpHttpSpanExporterBuilder();
+  }
+
+  /**
+   * Returns a new {@link OtlpHttpSpanExporter} using the default values.
+   *
+   * @return a new {@link OtlpHttpSpanExporter} instance.
+   */
+  public static OtlpHttpSpanExporter getDefault() {
+    return builder().build();
+  }
+
+  /** Shutdown the exporter. */
+  @Override
+  public CompletableResultCode shutdown() {
+    // TODO: should we update a boolean shutdown flag which prevents future exports?
+    final CompletableResultCode result = CompletableResultCode.ofSuccess();
+    this.spansSeen.unbind();
+    this.spansExportedSuccess.unbind();
+    this.spansExportedFailure.unbind();
+    return result;
+  }
+
+  interface RequestResponseHandler {
```

Comment:
Not in love with the name here, but this contains the behavior that varies based on whether JSON or Profobuf format is used.